frontend/src/components/Modal.js

```diff
@@ -20,13 +20,19 @@ const Modal = ({
   okButtonCss,
   cancelButtonCss,
   showTitleRequired,
+  secondaryOkButtonText,
+  onSecondaryOk,
```

Comment:
Nitpick: probably not "secondaryOk" but "secondaryAction", to me OK means acknowledge and dismiss the modal

Reply:
Agree sounds better.